In [ ]:
import numpy as np
import pandas as pd

To read the csv file using pandas we use read_csv.

pandas DataFrame is used to convert the csv file into data frame.

In [ ]:
df= pd.DataFrame(pd.read_csv("train.csv"))

In [ ]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

The top 5 rows are shown when head() fn is used.

In [ ]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

To find the dimensions of the data frame we use shape fn.

In [ ]:
df.shape

(891, 12)

In [ ]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Here it shows in Age column there are 177 missing values, Cabin column contains 687 missing values and Embarked column contains 2 missing values

#Now we are seperating the columns which have more than 40% of the values missing in the dataset

In [ ]:
#df.isnull().sum() returns a pandas series with columns name as the label index.
x=df.isnull().sum()
drop_col=x[x>(40/100 * df.shape[0])]

In [ ]:
drop_col

Cabin    687
dtype: int64

In [ ]:
drop_col.index

Index(['Cabin'], dtype='object')

In [ ]:
df.drop(drop_col.index, axis=1,inplace=True)
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

In [ ]:
df.fillna(df.mean(),inplace=True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

In [ ]:
df["Embarked"].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

For Embarked attribute, I fill the most frequent value ie "S" from the column

In [ ]:
df["Embarked"].fillna('S',inplace=True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

In [ ]:
df.corr()

PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.005007 -0.035144  0.033207 -0.057527 -0.001652   
Survived       -0.005007  1.000000 -0.338481 -0.069809 -0.035322  0.081629   
Pclass         -0.035144 -0.338481  1.000000 -0.331339  0.083081  0.018443   
Age             0.033207 -0.069809 -0.331339  1.000000 -0.232625 -0.179191   
SibSp          -0.057527 -0.035322  0.083081 -0.232625  1.000000  0.414838   
Parch          -0.001652  0.081629  0.018443 -0.179191  0.414838  1.000000   
Fare            0.012658  0.257307 -0.549500  0.091566  0.159651  0.216225   

                 Fare  
PassengerId  0.012658  
Survived     0.257307  
Pclass      -0.549500  
Age          0.091566  
SibSp        0.159651  
Parch        0.216225  
Fare         1.000000

I'm going to make a new column with SibSp(No.of sibilings/spouses aboard) and Parch(No.of Parents/Childrens aborad)

So,the new column is termed as FamilySize.

In [ ]:
df["FamilySize"]=df["SibSp"]+df["Parch"]
df.drop(["SibSp","Parch"],axis=1,inplace=True)

In [ ]:
df.corr()

PassengerId  Survived    Pclass       Age      Fare  FamilySize
PassengerId     1.000000 -0.005007 -0.035144  0.033207  0.012658   -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.069809  0.257307    0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.331339 -0.549500    0.065997
Age             0.033207 -0.069809 -0.331339  1.000000  0.091566   -0.248512
Fare            0.012658  0.257307 -0.549500  0.091566  1.000000    0.217138
FamilySize     -0.040143  0.016639  0.065997 -0.248512  0.217138    1.000000

By reading the above given correlation, we could sat that FamilySize in the ship  has very low correlance with survival rate.

So, now we'll check weather the person was alone  or not can affect the survival rate.

In [ ]:
df["Alone"]=[0 if df["FamilySize"] [i]>0 else 1 for i in df.index]

In [ ]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  \
0                            Braund, Mr. Owen Harris    male  22.0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
2                             Heikkinen, Miss. Laina  female  26.0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
4                           Allen, Mr. William Henry    male  35.0   

             Ticket     Fare Embarked  FamilySize  Alone  
0         A/5 21171   7.2500        S           1      0  
1          PC 17599  71.2833        C           1      0  
2  STON/O2. 3101282   7.9250        S           0      1  
3            113803  53.1000        S           1      0  
4            373450   8.0500        S           0      1

In [ ]:
df.groupby(["Alone"])["Survived"].mean()

Alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64

If the person is alone, they have less chance of surviving because the person with family would be prioritized over other people who are travelling alone.

In [ ]:
df[["Alone","Fare"]].corr()

Alone      Fare
Alone  1.000000 -0.271832
Fare  -0.271832  1.000000

So now we'll check if the person was not alone and the chance the ticket price is higher is high.

In [ ]:
df['Sex']=[0 if df["Sex"][i]=="male" else 1 for i in df.index]

In [ ]:
df.groupby(["Sex"])["Survived"].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

The above output shows female passengers have more chance of surviving than the male ones.

It shows women are prioritized over men.

In [ ]:
df.groupby(["Embarked"])["Survived"].mean()

Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

# CONCLUSION

Female passengers were prioritized over men.

People with high class tickets have higher survival rate than others.

Passengers who are travelling with their family have higher survival rate when compared with the person who are travelling alone.

Passengers who boarded the ship at Cherbourg, survived more in proportion than the others.